main/snprintf.c

```diff
@@ -143,8 +143,12 @@ PHPAPI char *php_gcvt(double value, int ndigit, char dec_point, char exponent, c
 {
 	char *digits, *dst, *src;
 	int i, decpt, sign;
+	int mode = ndigit >= 0 ? 2 : 0;
 
-	digits = zend_dtoa(value, 2, ndigit, &decpt, &sign, NULL);
+	if (mode == 0) {
+		ndigit = 17;
+	}
+	digits = zend_dtoa(value, mode, ndigit, &decpt, &sign, NULL);
```

Comment:
I have been checking that and haven't found any case that would be using EG(precision) and either 'f', 'F', 'e' or 'E' format which is the only place that calls php_conv_fp - the func that calls php_fcvt/php_ecvt. Of course I might have missed it so pls let me know if you are aware of any such case.